applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -231,48 +231,43 @@ static void search_start(void)
 {
 	int err;
 	struct location_config config;
-	enum location_method methods[3];
 	int methods_count = 0;
-	int methods_index_gnss = -1;
-	int methods_index_cellular = -1;
+	struct location_method_config methods_updated[CONFIG_LOCATION_METHODS_LIST_SIZE] = { 0 };
 
 	if (copy_cfg.no_data.neighbor_cell && copy_cfg.no_data.gnss && copy_cfg.no_data.wifi) {
 		SEND_EVENT(location, LOCATION_MODULE_EVT_DATA_NOT_READY);
 		LOG_ERR("All GNSS, cellular and Wi-Fi are configured off");
 		return;
 	}
 
-	if (!copy_cfg.no_data.gnss) {
-		methods[methods_count] = LOCATION_METHOD_GNSS;
-		methods_index_gnss = methods_count;
-		methods_count++;
-	}
-
-#if defined(CONFIG_LOCATION_METHOD_WIFI)
-	if (!copy_cfg.no_data.wifi) {
-		methods[methods_count] = LOCATION_METHOD_WIFI;
-		methods_count++;
-	}
-#endif
-
-	if (!copy_cfg.no_data.neighbor_cell) {
-		methods[methods_count] = LOCATION_METHOD_CELLULAR;
-		methods_index_cellular = methods_count;
-		methods_count++;
-	}
-
-	location_config_defaults_set(&config, methods_count, methods);
+	/* Set default location configuration configured at compile time */
+	location_config_defaults_set(&config, 0, NULL);
 
 	config.timeout = copy_cfg.location_timeout * MSEC_PER_SEC;
 
-	if (methods_index_gnss != -1) {
-		config.methods[methods_index_gnss].gnss.timeout =
-			DATA_FETCH_TIMEOUT_GNSS_SEARCH * MSEC_PER_SEC;
-	}
+	/* If any method type has been disabled, update the method list by using those methods
+	 * that are enabled
+	 */
+	if (copy_cfg.no_data.neighbor_cell || copy_cfg.no_data.gnss || copy_cfg.no_data.wifi) {
+		for (int i = 0; i < config.methods_count; i++) {
+			/* Use methods that are not disabled at run-time in no_data configuration */
+			if ((config.methods[i].method == LOCATION_METHOD_GNSS &&
+			     !copy_cfg.no_data.gnss) ||
+			    (config.methods[i].method == LOCATION_METHOD_WIFI &&
+			     !copy_cfg.no_data.wifi) ||
+			    (config.methods[i].method == LOCATION_METHOD_CELLULAR &&
+			     !copy_cfg.no_data.neighbor_cell)) {
+				memcpy(&methods_updated[methods_count],
```

Comment:
The Sonarlint extensions complains about out of bound access here. To me it seems fine.

Reply:
Is it because in the `for` loop there's no check whether `i` is greater than `CONFIG_LOCATION_METHODS_LIST_SIZE`? If `config.methods_count` would be greater than `CONFIG_LOCATION_METHODS_LIST_SIZE`, `memcpy()` could write past the end of `methods_updated`. I don't think `config.methods_count` can be greater than `CONFIG_LOCATION_METHODS_LIST_SIZE`, but SonarLint probably doesn't know that. 😃

Reply:
Yep, this is likely impossible to catch correctly by the static analysis tools.
I'll close this and get the PR merged.